src/mlpack/core/tree/address.hpp

```diff
@@ -0,0 +1,249 @@
+/**
+ * @file address.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * This file contains a series of functions for translating points to addresses
+ * and back and functions for comparing addresses.
+ *
+ * The notion of addresses is described in the following paper.
+ * @code
+ * @inproceedings{bayer1997,
+ *   author = {Bayer, Rudolf},
+ *   title = {The Universal B-Tree for Multidimensional Indexing: General
+ *       Concepts},
+ *   booktitle = {Proceedings of the International Conference on Worldwide
+ *       Computing and Its Applications},
+ *   series = {WWCA '97},
+ *   year = {1997},
+ *   isbn = {3-540-63343-X},
+ *   pages = {198--209},
+ *   numpages = {12},
+ *   publisher = {Springer-Verlag},
+ *   address = {London, UK, UK},
+ * }
+ * @endcode
+*/
+#ifndef MLPACK_CORE_TREE_ADDRESS_HPP
+#define MLPACK_CORE_TREE_ADDRESS_HPP
+
+namespace mlpack {
+
+namespace bound {
+
+namespace addr {
+
+/**
+ * Calculate the address of a point. Be careful, the point and the address
+ * variables should be equal-sized and the type of the address should correspond
+ * to the type of the vector.
+ *
+ * @param address The resulting address.
+ * @param point The point that is being translated to the address.
+ */
+template<typename AddressType, typename VecType>
+void PointToAddress(AddressType& address, const VecType& point)
```

Comment:
yeah, it's correct. In one dimensional case this transform preserves the ordering (lower addresses correspond lower points). The function looks like `DiscreteHilbertValue::CalculateValue()`. I introduced that since it is faster than recursive calculation (indeed, `RecursiveHilbertValue` was slower than `DiscreteHilbertValue`).